gateway/src/apicast/cli/command/start.lua

```diff
@@ -109,11 +109,28 @@ local function openresty_binary(candidates)
         find_openresty_command(candidates)
 end
 
+-- Assign different defaults to configuration_cache depending on
+-- configuration_loader. When it's lazy, we want 0 so it reloads on every
+-- request. However 0 it's incompatible with 'boot' mode, so we just keep
+-- it as nil.
+local function default_config_cache(options, context, configuration_loader)
+    local configuration_cache = options.cache or context.configuration_cache
+
+    if not configuration_cache and configuration_loader == 'lazy' then
```

Comment:
Can't we delegate this to the environment configurations ?

Reply:
Yes, that's better :+1: